### PR TITLE
Update app header

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
- Create WebSocket for host status in app header
- Visual style updates to move us closer to proposed UI
- Enabled https for devServer to allow secure WebSocket (wss)
- Sass loader config changes to allow sass variables to be
   used in single file components

### CODE REVIEW

**Comment by yoshiemuranaka** [comment]
waiting for #51 to be merged so I can reference event log store for server health status in header

**Review by derick-montague — APPROVED** [review]
Looks good. Can't seem to run on localhost anymore with the websocket, but can run on the IP that the dev server is running on for anyone else trying to test and for future work once this is merged in.

File: src/store/modules/GlobalStore.js
Comment (gtmills) [inline src/store/modules/GlobalStore.js:10]: Also could have an off transition, beyond the scope of this.

File: src/store/modules/GlobalStore.js
Comment (gtmills) [inline src/store/modules/GlobalStore.js:22]: how would we ever hit this case? 
HostState can only ever be running, off, or Quiesced, so we hit those first...

https://github.com/openbmc/phosphor-dbus-interfaces/blob/master/xyz/openbmc_project/State/Host.interface.yaml#L35

File: src/store/plugins/HostStatusPlugin.js
Comment (gtmills) [inline src/store/plugins/HostStatusPlugin.js]: A comment or 2 in here on *why* we are doing this could help when we upstream. Also maybe a link to the document that explains how all this works might also be helpful https://github.com/openbmc/docs/blob/b41aff0fabe137cdb0cfff584b5fe4a41c0c8e77/rest-api.md#event-subscription-protocol

File: src/store/modules/GlobalStore.js
Comment (gtmills) [inline src/store/modules/GlobalStore.js]: Why "Running"? but then we store it as "on"?

**Comment by gtmills** [comment]
Looks good! Thanks Yoshie! just had a couple of comments, if you fix up/answer, feel free to merge!

File: src/store/modules/GlobalStore.js
Comment (yoshiemuranaka) [reply src/store/modules/GlobalStore.js:22]: I added these transition states because they were being returned with the websocket events. I've seen transition.on and transition.reboot

File: src/store/modules/GlobalStore.js
Comment (yoshiemuranaka) [reply src/store/modules/GlobalStore.js]: Yeah, was thinking to change this. I was originally following the copy we have existing in our angular app, but with the new design proposals, it looks like the status will be indicated with the icon only and no text

File: src/store/plugins/HostStatusPlugin.js
Comment (yoshiemuranaka) [reply src/store/plugins/HostStatusPlugin.js]: Great documentation..hadn't seen this earlier. I will plan to add comments

File: src/store/modules/GlobalStore.js
Comment (yoshiemuranaka) [reply src/store/modules/GlobalStore.js:22]: I'm checking for either CurrentHostState or RequestedHostTransition in the websocket event

**Comment by derick-montague** [comment]
Migrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28729

**Pull request merged**
